### DISCUSSION TITLE
Enquiring regarding Carla's Add new sensors

### DISCUSSION THREAD

**Discussion by inv-samuel** [carla-simulator/carla#5671] [Q&A]
For adding a new sensor in Carla, The document mentions a path "Unreal/CarlaUE4/Plugins/Carla/Source/Carla/Sensor", Unfortunately, I can't find it in my system. The above-mentioned folder structure is present in the git but not in the locally installed software.  As per Carla's documentation for adding a new sensor they have mentioned building Carla from the source.  should I have to build from source to access these files?
I have follow these steps to install the Carla https://carla.readthedocs.io/en/latest/start_quickstart/

**Comment by ll7** [comment]
Yes, you have to build carla from source to add a new sensor type.

However, you don't need to build from source to attach an instance of an existing sensor type to your vehicle.

**Reply by inv-mary** [reply]
Hi
@ll7  when i was trying to add a safe distance sensor in carla (as per the doc : https://carla.readthedocs.io/en/0.9.13/tuto_D_create_sensor/) im facing an issue

WARNING: Version mismatch detected: You are trying to connect to a simulator that might be incompatible with this API
WARNING: Client API version = 0.9.13
WARNING: Simulator API version = 0.9.13-1-g8854804f4-dirty
INFO: Found the required file in cache! Carla/Maps/TM/Town10HD_Opt.bin
INFO: Found the required file in cache! Carla/Maps/Nav/Town10HD_Opt.bin
basic agent set destination
ActorBlueprint(id=sensor.other.safe_distance,tags=[sensor, other, safe_distance])
Assertion failed: px != 0, file C:\workspace\carla\Build\boost-1.72.0-install\include\boost/smart_ptr/shared_ptr.hpp, line 734

and the server crashed

Could you help us to figure out what we need to do? Thanks for your work and best regards.

**Reply by ll7** [reply]
I am not an expert, but it looks to me like you use the PythonAPI of your original, unmodified carla0.9.13 version in your python script. You have to use the `.egg` file of your modified carla version, that you build from source.

**Reply by ll7** [reply]
https://carla.readthedocs.io/en/latest/build_linux/#other-make-commands `make PythonAPI` is what you are looking for.
https://carla.readthedocs.io/en/latest/build_linux/#build-carla explains how to use the different `.egg` or `.whl` files.

**Comment by inv-mary** [comment]
hi,
@ll7 ,
how can i find town12 in carla map?

**Reply by ll7** [reply]
As far as I know, Town12 is not part of the official Carla0.9.13 release. However, there exists a prerelease version for Carla0.9.14 which is recommended for the Carla Leaderboard V2. Have a look at <https://leaderboard.carla.org/get_started/#11-download-the-carla-leaderboard-package>.

**Reply by inv-mary** [reply]
@ll7 thank you for your information

**Reply by inv-mary** [reply]
hi, @ll7 ,
when i tried to launch carla following error occurred.

The global shader cache file '/media/Disk1/CARLA_Leaderboard_20/Engine/GlobalShaderCache-GLSL_150_ES31.bin' is nissing
Your application is built to load COOKED content, No COOKED content was found: This usually means you did not cook content for this build.
It also may indicate missing cooked data for a shader platform(e.g. OpenGL under Windows): Make sure your platform's packaging settings include this Targeted RHI.
Alternatively build and run the UNCOOKED version instead,


and couldnt launch the server.How can fix this .Thanks in advance.

**Reply by ll7** [reply]
Sorry, I don't know.

**Reply by inv-samuel** [reply]
@inv-mary , This problem also occurs when vulkan utils is not installed. Installing vulkan-utils can help

sudo apt install vulkan-utils

**Reply by inv-mary** [reply]
@inv-samuel ,Thanks ..it worked
